Android/Self_Assessment_Devarshi/Advanced/app/src/main/java/com/example/advanced/FirstFragment.java

```diff
@@ -0,0 +1,100 @@
+package com.example.advanced;
+import com.example.advanced.api.ApiClient;
+import com.example.advanced.api.ApiInterface;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.advanced.models.Article;
+import com.example.advanced.models.News;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class FirstFragment extends Fragment {
+    public static final String API_KEY="588c2325670649f4949bf46584f71f73";
+    private RecyclerView mRecyclerView;
+    private TripAdapter mAdapter;
+    private RecyclerView.LayoutManager mLayoutManager;
+    private TransportViewModel transportViewModel;
+
+    private List<Article> articles=new ArrayList<>();
+    private ApiInterface apiInterface;
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             ViewGroup container, Bundle savedInstanceState) {
+        transportViewModel =
+                ViewModelProviders.of(this).get(TransportViewModel.class);
```

Comment:
Deprecated. Please use the constructors for ViewModelProvider directly.

Reply:
Done

Reply:
removed it entirely as it was not needed